### ISSUE TITLE
Error on Message.Builder subclass when running compileCommonMainKotlinMetadata

### ISSUE THREAD

**Issue by ebarrenechea** [square/wire#2707]
I'm trying to generate models for JVM, JS and iOS targets with `javaInterop = true` (I'm also using wire-moshi-adapter, which needs builders to work correctly). The plugin generates (seemingly correct) code but there's an error on `compileCommonMainKotlinMetadata` task where it errors out with the error below:
```
> Task :shared:compileCommonMainKotlinMetadata FAILED
e: file:///.../wiretest/shared/build/generated/source/wire/human/Person.kt:85:58 This class does not have a constructor
e: file:///.../wiretest/shared/build/generated/source/wire/human/Person.kt:230:65 This class does not have a constructor
```

It works correctly when `javaInterop = false` but then I run into issues with wire-moshi-adapter. I've noticed that it works correctly if I build the targets individually (e.g. only iOS) but fails whenever I toggle two platforms together (e.g. js and iOS). I've created a sample project which can reproduce the issue [here](https://github.com/ebarrenechea/wiretest).

**Comment by oldergod** [comment]
Thanks for reporting, the test case also helped a lot!

**Comment by SureshKumarA24** [comment]
We found a similar issue in the issue list and noticed that you fixed it in version 5.3.0. However, since we haven’t upgraded to Kotlin 2.0 yet, so we need this fix in the 4.8.1 build.

Please check: [GitHub Issue #2707](https://github.com/square/wire/issues/2707)
